src/main/java/org/codehaus/plexus/classworlds/realm/ClassRealm.java

```diff
@@ -218,7 +218,23 @@ private Class<?> unsynchronizedLoadClass(String name, boolean resolve) throws Cl
     @SuppressWarnings("Since15")
     protected Class<?> findClass(String moduleName, String name) {
         if (moduleName != null) {
-            return null;
+            // Delegate to parent for module-aware class loading to maintain proper module context
```

Comment:
I'm a bit surprised that the moduleName is not used anymore (only for the not-null check). It might be correct, but it looks very confusing, which is in general not a good sign